stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/messages/composer/MessageComposer.kt

```diff
@@ -418,23 +422,23 @@ internal fun DefaultComposerIntegrations(
         verticalAlignment = Alignment.CenterVertically
     ) {
         IconButton(
-            enabled = !hasCommandInput,
+            enabled = isAttachmentsButtonEnabled,
             modifier = Modifier
                 .size(32.dp)
                 .padding(4.dp),
             content = {
                 Icon(
                     painter = painterResource(id = R.drawable.stream_compose_ic_attachments),
                     contentDescription = stringResource(id = R.string.stream_compose_attachments),
-                    tint = if (hasCommandInput) ChatTheme.colors.disabled else ChatTheme.colors.textLowEmphasis,
+                    tint = if (isAttachmentsButtonEnabled) ChatTheme.colors.textLowEmphasis else ChatTheme.colors.disabled,
                 )
             },
             onClick = onAttachmentsClick
         )
 
-        val commandsButtonTint = if (hasCommandSuggestions && !hasTextInput) {
+        val commandsButtonTint = if (hasCommandSuggestions && !hasTextInput && !hasAttachments) {
             ChatTheme.colors.primaryAccent
-        } else if (!hasTextInput) {
+        } else if (!hasTextInput && !hasAttachments) {
```

Comment:
What do you think about extracting these 3 occurrences of `!hasTextInput && !hasAttachments` to `isCommandButtonEnabled`?

Reply:
Yes that is a good idea and will improve redability and save on unnecessary checks.